zerver/lib/test_classes.py

```diff
@@ -373,13 +374,12 @@ def submit_reg_form_for_user(
         """
         if full_name is None:
             full_name = email.replace("@", "_")
-
         payload = {
             'full_name': full_name,
             'password': password,
             'realm_name': realm_name,
             'realm_subdomain': realm_subdomain,
-            'key': find_key_by_email(email),
+            'key': key if key is not None else find_key_by_email(email),
```

Comment:
Do we need this? The test you're adding seems to be passing in a key.

Reply:
Just so that it does not break the other tests that are using this code so as to keep the original code intact and also if the key is passed then no need to call find_key_by_email() function which returns a None key value.

Reply:
Ohhh, I read this wrong, my bad. Yeah, I think the change makes sense.